cli/src/commands/split.rs

```diff
@@ -221,10 +221,17 @@ pub(crate) fn cmd_split(
     // Prompt the user to select the changes they want for the first commit.
     let target = select_diff(ui, &tx, &target_commit, &matcher, &diff_selector)?;
 
+    let legacy_change_id_behavior = tx.settings().get_bool("split.legacy-change-id-behavior")?;
```

Comment:
nit: remove the other `let legacy_change_id_behavior`

Reply:
I don't see any other `let legacy_change_id_behavior`. Did you confuse it with `let legacy_bookmark_behavior`? I did it when writing the code, and had some unexpected test failures :)

Reply:
> I don't see any other `let legacy_change_id_behavior`. Did you confuse it with `let legacy_bookmark_behavior`?

Oh, yes. Do we need two separate "legacy" options? I think they serve the same role. We have `legacy-bookmark-behavior` because in legacy mode, bookmarks (and the working copy) move forward to the second commit whereas the change id stays in the first. In non-legacy behavior, bookmarks should move/stay along with the change id.

Reply:
In my opinion, we only need this PR, without any config ;-)

Obviously not everyone agrees, so I'll probably step back and wait for a consensus on #3419